AssertionError is displayed as BROKEN in allure report insteaf of FAILED

*Environment (please complete the following information):**
 - **WebdriverIO version:** 7.0.8
 - **Mode:** WDIO
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 14.16
 - **NPM version:** 6.14.11
 - **Browser name and version:** Any
 - **Platform name and version:**  Windows 10
 - **Additional wdio packages used (if applicable):** "@wdio/cli": "^7.0.8",
    "@wdio/cucumber-framework": "^7.0.7",
    "@wdio/dot-reporter": "^7.0.7",
    "@wdio/local-runner": "^7.0.8",
    "@wdio/mocha-framework": "^7.0.7",
    "@wdio/spec-reporter": "^7.0.7",
    "@wdio/sync": "^7.0.8",
    "@wdio/allure-reporter": "^7.0.7",
     "expect": "^26.6.0",
    "expect-webdriverio": "^1.4.1",


**Describe the bug**
When an assertion failed with assert , Test status is BROKEN in allure-report xml file. 

**To Reproduce**

```
Feature: Test
My description feature

Scenario: Test
  Given I fail test

```

```
Given(
  'I fail test', () => {
    assert.isOk(false);
  }
);
```

**Expected behavior**
Test status should be FAILED in allure xml report instead of BROKEN
OK with wdio V6 -> Regression  (was working in `6.11.3`)

**wdio.config file:**
```
const expect = require('expect');
require('expect-webdriverio').setOptions({ wait: 30000 });
 framework: 'cucumber',

  reporters: ['spec',
    ['allure', {
      outputDir: 'report/allure-results',
      disableWebdriverStepsReporting: false,
      disableWebdriverScreenshotsReporting: false,
      useCucumberStepReporter: true,
    }]],
  cucumberOpts: {
    // <boolean> show full backtrace for errors
    backtrace: false,
    // <string[]> module used for processing required features
    requireModule: ['@babel/register'],
    // <boolean< Treat ambiguous definitions as errors
    failAmbiguousDefinitions: true,
    // <boolean> invoke formatters without executing steps
    // dryRun: false,
    // <boolean> abort the run on first failure
    failFast: false,
    // <boolean> Enable this config to treat undefined definitions as
    // warnings
    ignoreUndefinedDefinitions: false,
    // <string[]> ("extension:module") require files with the given
    // EXTENSION after requiring MODULE (repeatable)
    name: [],
    // <boolean> hide step definition snippets for pending steps
    snippets: true,
    // <boolean> hide source uris
    source: true,
    // <string[]> (name) specify the profile to use
    profile: [],
    // <string[]> (file/dir) require files before executing features
    require: [
      './src/steps/given.js',
      './src/steps/then.js',
      './src/steps/when.js',
      './src/steps/after.js',
      // Or search a (sub)folder for JS files with a wildcard
      // works since version 1.1 of the wdio-cucumber-framework
      // './src/**/*.js',
    ],
    // <string> specify a custom snippet syntax
    snippetSyntax: undefined,
    // <boolean> fail if there are any undefined or pending steps
    strict: true,
    // <string> (expression) only execute the features or scenarios with
    // tags matching the expression, see
    // https://docs.cucumber.io/tag-expressions/
    tagExpression: 'not @Pending',
    // <boolean> add cucumber tags to feature or scenario name
    tagsInTitle: true,
    // <number> timeout for step definitions
    timeout: 99999999,
  },
```

**Log**
Content of testsuite xml file
```
<?xml version='1.0'?>
<ns2:test-suite xmlns:ns2='urn:model.allure.qatools.yandex.ru' start='1614680398068' stop='1614680398491'>
    <name>My description feature</name>
    <title>My description feature</title>
    <test-cases>
        <test-case start='1614680398146' status='broken' stop='1614680398157'>
            <name>My description feature</name>
            <title>My description feature</title>
            <labels>
                <label name='language' value='javascript'/>
                <label name='framework' value='wdio'/>
                <label name='thread' value='0-0'/>
            </labels>
            <parameters>
                <parameter kind='argument' name='browser' value='firefox-85.0'/>
            </parameters>
            <steps>
                <step start='1614680398151' status='broken' stop='1614680398157'>
                    <name>Given I fail test with assert</name>
                    <title>Given I fail test with assert</title>
                    <attachments/>
                    <steps/>
                </step>
            </steps>
            <attachments>
                <attachment title='Screenshot' source='960a6fd5-98ff-47f7-8a8d-1c62ec156d2a-attachment.png' type='image/png' size='6711'/>
            </attachments>
            <failure>
                <message>AssertionError: expected false to be truthy</message>
                <stack-trace>AssertionError: expected false to be truthy
                    at World.&lt;anonymous> (C:\2-gitlab\end-to-end-tests-wdio\src\steps\/given.js:87:12)
                    at World.executeSync (C:\2-gitlab\end-to-end-tests-wdio\node_modules\@wdio\sync\build\index.js:38:22)
                    at C:\2-gitlab\end-to-end-tests-wdio\node_modules\@wdio\sync\build\index.js:69:68</stack-trace>
            </failure>
        </test-case>
    </test-cases>
</ns2:test-suite>
```

--- a/packages/wdio-allure-reporter/src/utils.ts
+++ b/packages/wdio-allure-reporter/src/utils.ts
@@ -19,18 +19,13 @@ export const getTestStatus = (test: TestStats | HookStats, config?: Options.Test
     }
 
     if (test.error) {
-        if (test.error.name && test.error.message) {
-            const message = test.error.message.trim()
-            return (test.error.name === 'AssertionError' || message.includes('Expect'))  ? 'failed' : 'broken'
+        if (test.error.message) {
+            const message = test.error.message.trim();
+            return (message.startsWith('AssertionError') || message.includes('Expect') || message.includes('expect')) ? 'failed' : 'broken';
         }
-
-        if (test.error.name) {
-            return test.error.name === 'AssertionError' ? 'failed' : 'broken'
-        }
-
         if (test.error.stack) {
-            const stackTrace = test.error.stack.trim()
-            return (stackTrace.startsWith('AssertionError') || stackTrace.includes('Expect'))  ? 'failed' : 'broken'
+            const stackTrace = test.error.stack.trim();
+            return (stackTrace.startsWith('AssertionError') || stackTrace.includes('Expect')) ? 'failed' : 'broken';
         }
     }
 

--- a/packages/wdio-allure-reporter/src/utils.ts
+++ b/packages/wdio-allure-reporter/src/utils.ts
@@ -20,12 +20,12 @@ export const getTestStatus = (test: TestStats | HookStats, config?: Options.Test
 
     if (test.error) {
         if (test.error.message) {
-            const message = test.error.message.trim();
-            return (message.startsWith('AssertionError') || message.includes('Expect') || message.includes('expect')) ? 'failed' : 'broken';
+            const message = test.error.message.trim()
+            return (message.startsWith('AssertionError') || message.includes('Expect') || message.includes('expect')) ? 'failed' : 'broken'
         }
         if (test.error.stack) {
-            const stackTrace = test.error.stack.trim();
-            return (stackTrace.startsWith('AssertionError') || stackTrace.includes('Expect')) ? 'failed' : 'broken';
+            const stackTrace = test.error.stack.trim()
+            return (stackTrace.startsWith('AssertionError') || stackTrace.includes('Expect')) ? 'failed' : 'broken'
         }
     }
 

--- a/packages/wdio-allure-reporter/src/utils.ts
+++ b/packages/wdio-allure-reporter/src/utils.ts
@@ -20,12 +20,12 @@ export const getTestStatus = (test: TestStats | HookStats, config?: Options.Test
 
     if (test.error) {
         if (test.error.message) {
-            const message = test.error.message.trim()
-            return (message.startsWith('AssertionError') || message.includes('Expect') || message.includes('expect')) ? 'failed' : 'broken'
+            const message = test.error.message.trim().toLowerCase()
+            return (message.startsWith('assertionerror') || message.includes('expect')) ? 'failed' : 'broken'
         }
         if (test.error.stack) {
-            const stackTrace = test.error.stack.trim()
-            return (stackTrace.startsWith('AssertionError') || stackTrace.includes('Expect')) ? 'failed' : 'broken'
+            const stackTrace = test.error.stack.trim().toLowerCase()
+            return (stackTrace.startsWith('assertionerror') || stackTrace.includes('expect')) ? 'failed' : 'broken'
         }
     }
 

--- a/packages/wdio-allure-reporter/tests/__fixtures__/cucumber.ts
+++ b/packages/wdio-allure-reporter/tests/__fixtures__/cucumber.ts
@@ -20,10 +20,10 @@ const suite = (type = 'feature') => ({
 } as any)
 
 const error = {
-    message: 'foo == bar',
+    message: 'AssertionError [ERR_ASSERTION]: foo == bar',
     stack: 'AssertionError [ERR_ASSERTION]: foo == bar',
     type: 'AssertionError [ERR_ASSERTION]',
-    name: 'AssertionError',
+    name: 'Error',
     expected: 'foo',
     actual: 'bar'
 }

--- a/packages/wdio-allure-reporter/tests/__fixtures__/testState.ts
+++ b/packages/wdio-allure-reporter/tests/__fixtures__/testState.ts
@@ -22,10 +22,10 @@ export function testPassed() {
 export function testFailed() {
     const error =
         {
-            message: 'foo == bar',
+            message: 'AssertionError [ERR_ASSERTION]: foo == bar',
             stack: 'AssertionError [ERR_ASSERTION]: foo == bar',
             type: 'AssertionError [ERR_ASSERTION]',
-            name: 'AssertionError',
+            name: 'Error',
             expected: 'foo',
             actual: 'bar'
         }

--- a/packages/wdio-allure-reporter/tests/utils.test.ts
+++ b/packages/wdio-allure-reporter/tests/utils.test.ts
@@ -26,7 +26,7 @@ describe('utils', () => {
 
         it('failed for AssertionError', () => {
             const config = { framework: 'mocha' }
-            const test = { error: { name: 'AssertionError' } }
+            const test = { error: { name: 'Error', message: 'AssertionError' } }
             expect(getTestStatus(test as any, config)).toEqual(testStatuses.FAILED)
         })
 

